xarray/core/dataset.py

```diff
@@ -1204,9 +1277,22 @@ def sel(self, method=None, tolerance=None, drop=False, **indexers):
         Dataset.isel_points
         DataArray.sel
         """
+        from .dataarray import DataArray
+
+        new_coords = {k: v._coords for k, v in indexers.items()
+                      if isinstance(v, DataArray)}
+
+        indexers = {k: v.variable if isinstance(v, DataArray) else v
```

Comment:
Rather than pass `xarray.Variable` objects to `remap_label_indexers`, can we just pass unlabeled numpy arrays? It's nice if we can simply internal functions so they don't need to worry about labels. (There is enough going on in the indexing routines already.)